src/DGtal/base/CowPtr.h

```diff
@@ -88,14 +89,19 @@ namespace DGtal
     const T& operator*()    const throw()   {return *myPtr;}
     const T* operator->()   const throw()   {return myPtr.get();}
     const T* get()          const throw()   {return myPtr.get();}
-    
+
+    template < typename U = T, typename std::enable_if< ! std::is_const<U>::value >::type* = nullptr >
```

Comment:
I think it was a very comprehensible line, especially the `typename U = T`:trollface: 
Ok, I will add a comment :wink: ( in doxygen style or just for the devs ? )